File: tvn-cosine.ai/tvn-cosine.ai/logic/fol/Connectors.cs
 namespace aima.core.logic.fol;

/**
 * @author Ravi Mohan
 * 
 */
public class Connectors {
	public static readonly string AND = "AND";

	public static readonly string OR = "OR";

	public static readonly string NOT = "NOT";

	public static readonly string IMPLIES = "=>";

	public static readonly string BICOND = "<=>";

	public static bool isAND(string connector) {
		return AND .Equals(connector);
	}

	public static bool isOR(string connector) {
		return OR .Equals(connector);
	}

	public static bool isNOT(string connector) {
		return NOT .Equals(connector);
	}

	public static bool isIMPLIES(string connector) {
		return IMPLIES .Equals(connector);
	}

	public static bool isBICOND(string connector) {
		return BICOND .Equals(connector);
	}
}

File: tvn-cosine.ai/tvn-cosine.ai.v2/logic/basic/firstorder/Connectors.java
namespace aima.core.logic.basic.firstorder;

/**
 * @author Ravi Mohan
 * @author Anurag Rai
 */
public class Connectors {
	public static final String AND = "&";

	public static final String OR = "|";

	public static final String NOT = "~";

	public static final String IMPLIES = "=>";

	public static final String BICOND = "<=>";

	public static bool isAND(String connector) {
		return AND.Equals(connector);
	}

	public static bool isOR(String connector) {
		return OR.Equals(connector);
	}

	public static bool isNOT(String connector) {
		return NOT.Equals(connector);
	}

	public static bool isIMPLIES(String connector) {
		return IMPLIES.Equals(connector);
	}

	public static bool isBICOND(String connector) {
		return BICOND.Equals(connector);
	}
}